### PR TITLE
Add support for Walleys Part Payment widget

### PR DESCRIPTION
**Pull request by MichaelBengtsson**
* Feature - Adds support for Walley Part Payment Widget on the cart and product pages in WooCommerce.

### CODE REVIEW

**Comment by MichaelBengtsson** [comment]
Task linked: [CU-86934uary Create package for (and add support for) Walley Part Payment Widget](https://app.clickup.com/t/86934uary)

File: classes/class-walley-checkout-assets.php [hunk 1]
```diff
@@ -261,4 +280,16 @@ public function enqueue_admin_metabox_scripts( $hook ) {
 		wp_enqueue_style( 'walley-metabox-css' );
 	}
 
-} new Walley_Checkout_Assets();
+	/**
+	 * Register the widget script for the part payment widget.
+	 */
+	public function register_widget_assets() {
```
Comment (NiklasHogefjord) [inline classes/class-walley-checkout-assets.php:286]: It looks as these files are enqueued on all pages. Should we include a check so they only are enqueued on product pages if activated in settings?

File: collector-checkout-for-woocommerce.php [hunk 2]
```diff
@@ -138,6 +149,7 @@ public function init() {
 			include_once COLLECTOR_BANK_PLUGIN_DIR . '/classes/class-walley-checkout.php';
 			include_once COLLECTOR_BANK_PLUGIN_DIR . '/classes/class-walley-checkout-confirmation.php';
 			include_once COLLECTOR_BANK_PLUGIN_DIR . '/classes/class-walley-checkout-session.php';
+			include_once COLLECTOR_BANK_PLUGIN_DIR . '/classes/class-walley-part-payment-widget.php';
```
Comment (NiklasHogefjord) [inline collector-checkout-for-woocommerce.php:152]: What happens if the old checkout API is used? Will it still display the part payment widget settings?

**Review by NiklasHogefjord — CHANGES_REQUESTED** [review]

File: classes/class-walley-checkout-assets.php (hunk 1 above)
Comment (MichaelBengtsson) [reply classes/class-walley-checkout-assets.php:286]: This part of the code only registers the assets. They are not enqueued here.

That is done in the package for the widgets themselves. The Cart and Product widgets both have a method that's called `is_valid_page` where they check for `is_cart` and `is_product` respectively.
Before we enqueue the scripts we check against this method and if the widget itself is enabled by using the setting. We only register the scripts this early to ensure they are available when the package needs to enqueue them.

We pass the handles to the Cart and Product widget classes when we register them in the [Walley_Part_Payment_Widget class](https://github.com/krokedil/collector-checkout-for-woocommerce/blob/develop-part-payment-widget/classes/class-walley-part-payment-widget.php#L58-L74). These handles are then enqueued when the classes for the widgets notice that the widget should be used.

I cant see that the scripts or style are enqueued on any other page then what they are intended for in my testing.

File: collector-checkout-for-woocommerce.php (hunk 2 above)
Comment (MichaelBengtsson) [reply collector-checkout-for-woocommerce.php:152]: Yes that is correct. Ill fix that and commit a change to only include the file and initiate the class when using the new api

**Pull request merged**